scipy/signal/spectral.py

```diff
@@ -511,11 +511,423 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+    reconstruction.
+
+    Some examples of windows that satisfy COLA:
+        - Rectangular window at 0% & 50% overlap
+        - Bartlett window at 50% overlap
+        - Hanning window at 50% & 75% overlap
+        - Any Blackman family window at 2/3 overlap
+        - Any window with noverlap = nperseg-1
+
+    .. versionadded:: 0.18.0
+
+    References
+    ----------
+    .. [1] Julius O. Smith III, "Spectral Audio Signal Processing", W3K
+           Publishing, 2011,ISBN 978-0-9745607-3-1.
+    """
+
+    nperseg = int(nperseg)
+    if nperseg < 1:
+        raise ValueError('nperseg must be a positive integer')
+
+    if noverlap >= nperseg:
+        raise ValueError('noverlap must be less than nperseg.')
+    noverlap = int(noverlap)
+
+    if isinstance(window, string_types) or type(window) is tuple:
+        win = get_window(window, nperseg)
+    else:
+        win = np.asarray(window)
+        if len(win.shape) != 1:
+            raise ValueError('window must be 1-D')
+        if win.shape[0] != nperseg:
+            raise ValueError('window must have length of nperseg')
+
+    step = nperseg - noverlap
+    binsums = np.sum((win[ii*step:(ii+1)*step] for ii in range(nperseg//step)),
+                      axis=0)
+
+    if nperseg % step != 0:
+        binsums[:nperseg % step] += win[-(nperseg % step):]
+
+    deviation = binsums - np.median(binsums)
+    return np.max(np.abs(deviation)) < tol
+
+
+def stft(x, fs=1.0, window='hann', nperseg=256, noverlap=None, detrend=False,
+         padded=True, centered=True, strict=True, axis=-1):
+    """
+    Compute the Short Time Fourier Transform (STFT).
 
-    return freqs, time, Pxy
+    STFTs can be used as a way of quantifying the change of a nonstationary
+    signal's frequency and phase content over time.
+
+    Parameters
+    ----------
+    x : array_like
+        Time series of measurement values
+    fs : float, optional
+        Sampling frequency of the `x` time series. Defaults to 1.0.
+    window : str or tuple or array_like, optional
+        Desired window to use. See `get_window` for a list of windows and
+        required parameters. If `window` is array_like it will be used
+        directly as the window and its length will be used for nperseg.
+        Defaults to a Hann window.
+    nperseg : even int, optional
+        Length of each segment. Defaults to 256. Must be an even number.
+    noverlap : int, optional
+        Number of points to overlap between segments. If None, ``noverlap =
+        nperseg // 2``. Defaults to None. When specified, the COLA
+        constraint must be met (see Notes below).
+    detrend : str or function or False, optional
+        Specifies how to detrend each segment. If `detrend` is a string, it
+        is passed as the ``type`` argument to `detrend`. If it is a
+        function, it takes a segment and returns a detrended segment. If
+        `detrend` is False, no detrending is done. Defaults to False.
+    padded : bool, optional
```

Comment:
It's not clear whether this pads both the beginning and the end, or just the end